File: C#/1-two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        int[] indices = new int[2];
        for(int i = 0; i < nums.Length; i++) {
            bool found = false;
            for(int j = i + 1; j < nums.Length; j++) {
                if(nums[i] + nums[j] == target) {
                    indices[0] = i;
                    indices[1] = j;
                    found = true;
                    break;
                }
            }
            if(found) {
                break;
            }
        }
        return indices;
    }
}

File: Java/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        int[] result = new int[2];
        
        // build set while scanning:
        for(int i = 0; i < nums.length; i++) {
            // compare diffs with set elements
            int current = nums[i];
            int diff = target - current;
            if(map.containsKey(diff)) {
                result[1] = i;
                result[0] = map.get(diff);
                break;
            } else {
                map.put(current, i);
            }
        }

        // return result
        return result;
    }
}